File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for n in nums:
            i = abs(n) - 1
            if nums[i] > 0:
                nums[i] *= -1
        return [i + 1 for i in range(len(nums)) if nums[i] > 0]

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.go
func findDisappearedNumbers(nums []int) []int {
    ans := []int{}
    for _, x := range nums {
        i := abs(x) - 1
        if nums[i] > 0 {
            nums[i] = -nums[i]
        }
    }
    for i := 0; i < len(nums); i++ {
		if nums[i] > 0 {
			ans = append(ans, i+1)
		}
	}
	return ans
}
func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
